src/internal/mod.rs

```diff
@@ -477,6 +485,52 @@ impl<'de> Deserialize<'de> for PrivateKey {
     }
 }
 
+impl PrivateKey {
+    /// Augment this private key with another, producing a new PrivateKey
+    fn augment(&self, augmenting_key: &AugmentationFactor) -> Result<PrivateKey, IronOxideErr> {
+        use recrypt::Revealed;
+        let zero: RecryptPrivateKey = RecryptPrivateKey::new([0u8; 32]);
+        if Revealed((augmenting_key.clone().0).0) == Revealed(zero.clone()) {
+            Err(IronOxideErr::UserPrivateKeyRotationError(
+                "Augmenting key cannot be zero".into(),
+            ))
+        } else {
+            // this subtraction needs to be the additive inverse of what the service is doing
+            let augmented_key = self.0.clone() - augmenting_key.clone().into();
```

Comment:
Could have compared `self` and `augmenting_key` and errored if they were the same, before doing the subtraction. But with the coming change to do this in Fr instead of Fp, you would have to do that comparison after converting self and augmenting_key to Fr elements.